### DISCUSSION TITLE
Why tracing needs to be manually set up on the application?

### DISCUSSION THREAD

**Discussion by williamoliveira** [linkerd/linkerd2#5000] [General] [answered]
The docs says you need to do all the tracing work by hand: https://linkerd.io/2/tasks/distributed-tracing/
Why cant it be like istio.io where the proxy takes care of it all? https://istio.io/latest/faq/distributed-tracing/

Is there a way to make it easier?

**Comment by grampelberg** [comment, marked as answer]
@williamoliveira Istio's tracing requires the same modifications (see the first paragraph on your link). There's no way around requiring applications to forward tracing headers.

**Reply by grampelberg** [reply]
Also, see https://istio.io/latest/faq/distributed-tracing/#istio-copy-headers for more details.
